src/main/resources/assets/computercraft/lua/rom/programs/monitor.lua

```diff
@@ -39,15 +39,25 @@ local function resume( ... )
     return param
 end
 
+local timers = {}
+
 local ok, param = pcall( function()
     local sFilter = resume()
     while coroutine.status( co ) ~= "dead" do
         local tEvent = table.pack( os.pullEventRaw() )
+        if tEvent[1] == "timer" and timers[tEvent[2]] then
```

Comment:
Instead of hijacking the event, I wonder if it'd be better just to do what we do elsewhere and fire another event. So something like:

```lua
if coroutine.status( co ) ~= "dead" and (sFilter == nil or sFilter == "mouse_up") then
  if tEvent[1] == "timer" and timers[tEvent[2]] then
    sFilter = resume ("mouse_up", -- etc...
```

While we're technically sending an event the receiving coroutine will never use, it is a bit more consistent with the other handlers.

Reply:
Then keep the old timer event as well I assume?

Reply:
Yep.

Reply:
Oh, can you also move it to below the initial resume, so it's with the other events? Sorry!

Reply:
You'll also need to include the event filtering code, like the other cases.

Reply:
I believe that's done now.  Anything I missed?